python/private/py_console_script_binary.bzl

```diff
@@ -0,0 +1,83 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implementation for the macro to generate an console_script py_binary from reading the 'entry_points.txt'.
+"""
+
+load("//python:py_binary.bzl", "py_binary")
+load(":py_console_script_gen.bzl", "py_console_script_gen")
+
+def _dist_info(pkg):
+    """Return the first candidate for the dist_info target label.
+
+    We cannot do `Label(pkg)` here because the string will be evaluated within
```

Comment:
Similarly, if a user passes Label() (because they need for force resolution within their context for bzlmod), won't this code fail? Because it's expecting a str arg, not a Label object.

NBD for now. Just put a TODO and we can fix in a follow up PR

Reply:
Creating a ticket and leaving a TODO was more time than fixing the case to handle if the user passes the `Label` imho, so included the fix here.